zerver/migrations/0436_realmauthenticationmethods.py

```diff
@@ -11,12 +11,13 @@ def fill_RealmAuthenticationMethod_data(
 ) -> None:
     Realm = apps.get_model("zerver", "Realm")
     RealmAuthenticationMethod = apps.get_model("zerver", "RealmAuthenticationMethod")
+    rows_to_create = []
     for realm in Realm.objects.order_by("id"):
-        rows_to_create = []
         for key, value in realm.authentication_methods.iteritems():
             if value:
                 rows_to_create.append(RealmAuthenticationMethod(name=key, realm_id=realm.id))
-        RealmAuthenticationMethod.objects.bulk_create(rows_to_create)
+
+    RealmAuthenticationMethod.objects.bulk_create(rows_to_create, batch_size=10000)
```

Comment:
I think using a `batch_size` is still good in these kinds of migrations to avoid the possibility of the opposite issue with making one excessively massive query?

Reply:
Yes, that's a great pattern, though probably Zulip Cloud is the only place where it'll matter for this particular migration.